modules/c++/xml.lite/include/xml/lite/Speed.h

```diff
@@ -0,0 +1,341 @@
+     * manipulate it. A reasonable use case would be to parse an XML template
+     * file and fill it out or possibly add to it. If the goal is gathering
+     * information for processing elsewhere, consider using XPath or XQuery,
+     * available in libxml2. CODA provides libxml2 as an optional driver for
+     * XML functionality.
+     *
+     * \param doc Pointer to the xml::lite::Document to now refer to.
+     * \param steal Steal ownership of the root element of doc.
+     *              Default is true.
+     */
+    void fromDocument(xml::lite::Document *doc, bool steal=true);
+
+    /*!
+     * \func setCDATA
+     * \brief Set the value of a CDATA element at at location
+     *
+     *  setCDATA takes a string value and inserts it at the given location
+     *  surrounded by the proper XML syntax: '<![CDATA[' before and ']]>'after.
+     *  CDATA elements are not parsed as part of the XML document and so are
+     *  suitable for embedding arbitrary strings, e.g. HTML content.
+     *
+     *  CDATA sections are treated by this library in the same way as text
+     *  content of an element. There are limitations inherent in this design
+     *  which are meant not to be confining. While it is conceptually possible
+     *  to have both text and CDATA, or multiple CDATAs possibly interspersed
+     *  with text, etc. in the same tag, this function will not support that. A
+     *  call to setCDATA will overwrite any previous text value or CDATA at
+     *  that location. In these cases, compose the complete string and then use
+     *  setData.
+     *
+     * \param elname The full name of the element.
+     * \param cdata  The contents of the CDATA section.
+     */
+    void setCDATA(const std::string &elname, const std::string &cdata);
+    // TODO: 
+    // setComment(const std::string &elname, const std::string &cdata);
+    // declareSchema(const std::string &elname,
+    //               const std::string &nsPrefix,
+    //               const std::string &nsURI);
+
+  protected:
+    /*! \func buildTOC
+     *  \brief Build the table of contents map
+     *
+     */
+    void buildTOC(const std::string &parent="",
+                  xml::lite::Element const * element=NULL);
+
+    struct KeyParts
+    {
+      std::string proper_name;
+      std::string parent;
+      std::string child;
+      std::string child_name;
+      size_t      child_ord;
+    };
+    
+    KeyParts vivify(const std::string &elname);
+    KeyParts extractParts(const std::string &longname);
+
+    //! The table of contents for the XML document.
+    std::map<std::string, xml::lite::Element*> mTOC;
+    //! The namespaces known to exist in the document.
+    std::map<std::string, std::string> mNamespaces;
+
+};
+} // end namespace "lite"
+} // end namespace "xml"
+//-----------------------------------------------------------------
+  template <typename T>
+void xml::lite::Speed::setData(const std::string &elname, const T &value)
```

Comment:
Put all these methods inside the namespace rather than putting them outside... then you can remove all the `xml::lite::` business